frontend/src/App.tsx

```diff
@@ -96,6 +103,39 @@ function App() {
   const auth = useAuth();
   const routes = auth.isAuthenticated ? authenticatedRoutes : publicRoutes;
   const RouterElement = useRoutes(routes);
+  const { search } = useLocation();
+  const searchParams = new URLSearchParams(search);
+  const tenantSlug = searchParams.get("tenantSlug") || "";
+  const authToken = searchParams.get("authToken") || "";
+  const redirectTo = searchParams.get("redirectTo") || "";
+  const attemptLogin = !!(tenantSlug && authToken);
+  const [attemptingLogin, setAttemptingLogin] = useState(attemptLogin);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (attemptLogin) {
+      const authConfig: AuthConfig = {
+        tenantSlug: tenantSlug,
+        authToken: authToken,
+      };
+      auth.login(authConfig, false).then((isValidLogin) => {
+        setAttemptingLogin(false);
+        if (isValidLogin) {
+          navigate(redirectTo);
```

Comment:
When navigating to a different route, it would be better to replace the current route with the new one in the browser's history instead of just adding the new route to the browser's history.
The reason is that if the user pushed the Back button of the browser to go back in history, it would reach again this route where the login is attempted and the user is automatically redirected to a new route. So the user wouldn't be able to go back in history, always being redirected to the latest page.

```suggestion
          navigate(redirectTo, { replace: true });
```